src/compiler/core.ts

```diff
@@ -109,6 +109,17 @@ namespace ts {
         return undefined;
     }
 
+    /** Works like Array.prototype.find, returning `undefined` if no element satisfying the predicate is found. */
+    export function tryFind<T>(array: T[], predicate: (element: T, index: number) => boolean): T | undefined {

```

Comment:
It's the equivalent of bind (`>>=`) or something like `Data.Foldable.concatMap` except that it takes `Array<T>` to `Exception<T>`. So, yeah, `findMap` is not bad, or `mapFirst` or `mapSingle`.